R/drive_mkdir.R

```diff
@@ -9,13 +9,27 @@
 drive_mkdir <- function(dir = NULL, path = NULL, verbose = TRUE) {
   parent <- NULL
 
-  ## TO DO: accept path input in the generalized "Drive file" style
-  ## and check that it's a folder
   if (!is.null(path)) {
-    path <- append_slash(path)
-    parent <- get_one_path(path)$id
+    if (inherits(path, "dribble") || inherits(path, "drive_id")) {
+      path <- as_dribble(path)
+      if (!is_folder(path)){
+        stop(
+          glue::glue_data(path, "'path' is not a folder:\n{name}"),
+          call. = FALSE
+        )
+      }
+    } else {
+      path <- append_slash(path)
+      path <- get_one_path(path)
+      if (path$mimeType != "application/vnd.google-apps.folder") {
```

Comment:
@jennybc do you think this makes sense? this keeps the general framework of the functions & adds in a check for whether the input is a `dribble` or `drive_id`. I think an alternate way to do it would just be to do `as_dribble()`, check `is_folder()` and then make sure it only identifies 1 row. This eliminates the `get_one_path()` utility, though, and I was worried I'm not thinking of something important that that function does.